packages/synapse-react-client/src/components/AclEditor/AclEditorTestUtils.ts

```diff
@@ -2,12 +2,65 @@ import userEvent from '@testing-library/user-event'
 import { screen, waitFor, within } from '@testing-library/react'
 import { REMOVE_BUTTON_LABEL } from './ResourceAccessItem'
 
+/**
+ * Find a row in the ACL editor that contains the specified principal name.
+ * @param rows
+ * @param principalName
+ */
+export function queryForRowWithPrincipalName(
+  rows: HTMLElement[],
+  principalName: string,
+) {
+  return rows.find(row => {
+    return within(row).queryByText(principalName, { exact: false })
+  })
+}
+
+/**
+ * Verify that a row in the ACL editor contains the expected principal name and access type.
+ * @param row
+ * @param principalName
+ * @param accessTypeLabel
+ */
+export async function confirmItemViaQuery(
+  rows: HTMLElement[],
+  principalName: string,
+  accessTypeLabel: string,
+) {
+  let row: HTMLElement | undefined
+  try {
+    await waitFor(() => {
+      row = queryForRowWithPrincipalName(rows, principalName)!
+      expect(row).toBeInTheDocument()
+    })
+  } catch (e) {
+    screen.debug()
+    throw new Error(`Principal ${principalName} not found in ACL`, e)
+  }
+
+  const editorCombobox = within(row!).queryByRole('combobox')
+  if (editorCombobox) {
+    expect(editorCombobox).toHaveTextContent(accessTypeLabel)
+  } else {
+    within(row!).getByText(accessTypeLabel)
+  }
+  return row!
+}
+
+/**
+ * Verify that a row in the ACL editor contains the expected principal name and access type.
+ * @param row
+ * @param principalName
+ * @param accessTypeLabel
+ */
 export function confirmItem(
   row: HTMLElement,
   principalName: string,
   accessTypeLabel: string,
 ) {
-  expect(within(row).getByRole('link')).toHaveTextContent(principalName)
+  expect(within(row).queryByRole('link') ?? row).toHaveTextContent(
+    principalName,
+  )
```

Comment:
Public/authenticated groups don't have links, so we have to adjust this selector